### DISCUSSION TITLE
SD.Next Release 2025-07-29

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/sdnext#4079] [Announcements]
## SD.Next Release 2025-07-29

This is a big one: simply looking at number of changes, probably the biggest release since the project started!  

Feature highlights include:  
- [ModernUI](https://github.com/user-attachments/assets/6f156154-0b0a-4be2-94f0-979e9f679501) has quite some redesign which should make it more user friendly and easier to navigate plus several new UI themes  
  If you're still using **StandardUI**, give [ModernUI](https://vladmandic.github.io/sdnext-docs/Themes/) a try!  
- New models such as [WanAI 2.2](https://wan.video/) in 5B and A14B variants for both *text-to-video* and *image-to-video* workflows as well as *text-to-image* workflow!  
  and also [FreePix F-Lite](https://huggingface.co/Freepik/F-Lite), [Bria 3.2](https://huggingface.co/briaai/BRIA-3.2) and [bigASP 2.5](https://civitai.com/models/1789765?modelVersionId=2025412)  
- Redesigned [Video](https://vladmandic.github.io/sdnext-docs/Video) interface with support for general video models plus optimized [FramePack](https://vladmandic.github.io/sdnext-docs/FramePack) and [LTXVideo](https://vladmandic.github.io/sdnext-docs/LTX) support  
- Fully integrated nudity detection and optional censorship with [NudeNet](https://vladmandic.github.io/sdnext-docs/NudeNet)  
- New background replacement and relightning methods using **Latent Bridge Matching** and new **PixelArt** processing filter  
- Enhanced auto-detection of default sampler types/settings results in avoiding common mistakes  
- Additional **LLM/VLM** models available for captioning and prompt enhance  
- Number of workflow and general quality-of-life improvements, especially around **Styles**, **Detailer**, **Preview**, **Batch**, **Control**  
- Compute improvements  
- [Wiki](https://github.com/vladmandic/automatic/wiki) & [Docs](https://vladmandic.github.io/sdnext-docs/) updates, especially new end-to-end [Parameters](https://vladmandic.github.io/sdnext-docs/Parameters/) page  

In this release we finally break with legacy with the removal of the original [A1111](https://github.com/AUTOMATIC1111/stable-diffusion-webui/) codebase which has not been maintained for a while now  
This plus major cleanup of codebase and external dependencies resulted in ~55k LoC (*lines-of-code*) reduction and spread over [~750 files](https://github.com/vladmandic/sdnext/pull/4017) in ~200 commits!  

We also switched project license to [Apache-2.0](https://github.com/vladmandic/sdnext/blob/dev/LICENSE.txt) which means that SD.Next is now fully compatible with commercial and non-commercial use and redistribution regardless of modifications!  

And (*as always*) many bugfixes and improvements to existing features!  
For details, see [ChangeLog](https://github.com/vladmandic/automatic/blob/master/CHANGELOG.md)  

> [!NOTE]  
> We recommend clean install for this release due to sheer size of changes  
> Although upgrades and existing installations are tested and should work fine!  

![Screenshot](https://github.com/user-attachments/assets/6f156154-0b0a-4be2-94f0-979e9f679501)

[ReadMe](https://github.com/vladmandic/automatic/blob/master/README.md) | [ChangeLog](https://github.com/vladmandic/automatic/blob/master/CHANGELOG.md) | [Docs](https://vladmandic.github.io/sdnext-docs/) | [WiKi](https://github.com/vladmandic/automatic/wiki) | [Discord](https://discord.com/invite/sd-next-federal-batch-inspectors-1101998836328697867)

### Details

- **License**  
  - SD.Next [license](https://github.com/vladmandic/sdnext/blob/dev/LICENSE.txt) switched from **aGPL-v3.0** to **Apache-v2.0**  
    this means that SD.Next is now fully compatible with commercial and non-commercial use and redistribution regardless of modifications!  
- **Models**
  - [WanAI Wan 2.2](https://github.com/Wan-Video/Wan2.2) both 5B and A14B variants, for both T2V and I2V support  
    go to: *video -> generic -> wan -> pick variant*  
    optimized support with *VACE*, etc. will follow soon  
    *caution* Wan2.2 on its own is ~68GB, but also includes optional second-stage for later low-noise processing which is absolutely massive at additional ~54GB  
    you can enable second stage processing in *settings -> model options*, its disabled by default  
    *note*: quantization and offloading are highly recommended regardless of first-stage only or both stages!  
  - [WanAI Wan](https://wan.video/) T2V models for T2I workflows  
    Wan is originally designed for *video* workflows, but now also be used for *text-to-image* workflows!  
    supports *Wan-2.1 in 1.3B* and 14B variants and *Wan-2.2 in 5B and A14B* variants  
    supports all standard features such as quantization, offloading, TAESD preview generation, LoRA support etc.  
    can also load unet/transformer fine-tunes in safetensors format using UNET loader  
    simply select in *networks -> models -> reference*  
    *note* 1.3B model is a bit too small for good results and 14B is very large at 78GB even without second-stage so aggressive quantization and offloading are recommended  
  - [FreePix F-Lite](https://huggingface.co/Freepik/F-Lite) in *7B, 10B and Texture* variants  
    F-Lite is a 7B/10B model trained exclusively on copyright-safe and SFW content, trained on internal dataset comprising approximately 80 million copyright-safe images  
    available via *networks -> models -> reference*  
  - [Bria 3.2](https://huggingface.co/briaai/BRIA-3.2)  
    Bria is a smaller 4B parameter model built entirely on licensed data and safe for commercial use  
    *note*: this is a gated model, you need to [accept terms](https://huggingface.co/briaai/BRIA-3.2) and set your [huggingface token](https://vladmandic.github.io/sdnext-docs/Gated/)  
    available via *networks -> models -> reference*  
  - [bigASP 2.5](https://civitai.com/models/1789765)  
    bigASP is an experimental SDXL finetune using Flow matching method  
    load as usual, and leave sampler set to *Default*  
    or you can use following samplers: *UniPC, DPM, DEIS, SA*  
    required sampler settings: *prediction-method=flow-prediction*, *sigma-method=flowmatch*  
    recommended sampler settings: *flow-shift=1.0*  
  - [LBM: Latent Bridge Matching](https://github.com/gojasper/LBM)  
    very fast automatic image background replacement methods with relightning!  
    *simple*: automatic background replacement using [BiRefNet](https://github.com/ZhengPeng7/BiRefNet)  
    *relighting*: automatic background replacement with reglighting so source image fits desired background  
    with optional composite blending  
    available in *img2img or control -> scripts*  
  - add **FLUX.1-Kontext-Dev** inpaint workflow  
  - add **FLUX.1-Kontext-Dev** **Nunchaku** support  
    *note*: FLUX.1 Kontext is about 2-3x faster with Nunchaku vs standard execution!  
  - support **FLUX.1** all-in-one safetensors  
  - support for [Google Gemma 3n](https://huggingface.co/google/gemma-3n-E4B-it) E2B and E4B LLM/VLM models  
    available in **prompt enhance** and process **captioning**  
  - support for [HuggingFace SmolLM3](https://huggingface.co/HuggingFaceTB/SmolLM3-3B) 3B LLM model  
    available in **prompt enhance**  
  - add [fal AuraFlow 0.2](https://huggingface.co/fal/AuraFlow-v0.2) in addition to existing [fal AuraFlow 0.3](https://huggingface.co/fal/AuraFlow-v0.3) due to large differences in model behavior  
    available via *networks -> models -> reference*  
  - add integrated [NudeNet](https://vladmandic.github.io/sdnext-docs/NudeNet) as built-in functionality  
    *note*: used to be available as a separate [extension](https://github.com/vladmandic/sd-extension-nudenet)  
- **Video**
  - redesigned **Video** interface  
  - support for **Generic** video models  
    includes support for many video models without specific per-model optimizations  
    included: *Hunyuan, LTX, WAN, Mochi, Latte, Allegro, Cog*  
    supports quantization, offloading, frame interpolation, etc.  
  - support for optimized [FramePack](https://vladmandic.github.io/sdnext-docs/FramePack)  
    with *t2i, i2i, flf2v* workflows  
    LoRA support, prompt enhance, etc.  
    now fully integrated instead of being a separate extension  
  - support for optmized [LTXVideo](https://vladmandic.github.io/sdnext-docs/LTX)  
    with *t2i, i2i, v2v* workflows  
    optional native upsampling and video refine workflows  
    LoRA support with different conditioning types such as Canny/Depth/Pose, etc.  
  - support for post load quantization  
- **UI**  
  - major update to modernui layout  
  - add new Windows-like *Blocks* UI theme  
  - redesign of the *Flat* UI theme  
  - enhanced look&feel for *Gallery* tab with better search and collapsible sections, thanks to @CalamitousFelicitousness
- **WIKI**  
  - new [Parameters](https://vladmandic.github.io/sdnext-docs/Parameters/) page that lists and explains all generation parameters  
    massive thanks to @CalamitousFelicitousness for bringing this to life!  
  - updated *Models, Video, LTX, FramePack, Styles*, etc.
- **Compute**  
  - support for [SageAttention2++](https://github.com/thu-ml/SageAttention)  
    provides 10-15% performance improvement over default SDPA for transformer-based models!  
    enable in *settings -> compute settings -> sdp options*  
    *note*: SD.Next will use either SageAttention v1/v2/v2++, depending which one is installed  
    until authors provide pre-build wheels for v2++, you need to install it manually or SD.Next will auto-install v1  
  - support for `torch.compile` for LLM: captioning/prompt-enhannce  
  - support for `torch.compile` with repeated-blocks  
    reduces time-to-compile 5x without loss of performance!  
    enable in *settings -> model compile -> repeated*  
    *note*: torch.compile is not compatible with balanced offload  
- **Other**  
  - **Styles** can now include both generation params and server settings  
    see [Styles docs](https://vladmandic.github.io/sdnext-docs/Styles/) for details  
  - **TAESD** is now default preview type since its the only one that supports most new models  
  - support **TAESD** preview and remote VAE for **HunyuanDit**  
  - support **TAESD** preview and remote VAE for **AuraFlow**  
  - support **TAESD** preview for **WanAI**  
  - SD.Next now starts with *locked* state preventing model loading until startup is complete  
  - warn when modifying legacy settings that are no longer supported, but available for compatibilty  
  - warn on incompatible sampler and automatically restore default sampler  
  - **XYZ grid** can now work with control tab:  
    if controlnet or processor are selected in xyz grid, they will overwrite settings from first unit in control tab,  
    when using controlnet/processor selected in xyz grid, behavior is forced as control-only  
    also freely selectable are control strength, start and end values  
  - **Batch** warn on unprocessable images and skip operations on errors so that other images can still be processed  
  - **Metadata** improved parsing and detect foreign metadata  
    detect ComfyUI images  
    detect InvokeAI images  
  - **Detailer** add `expert` mode where list of detailer models can be converted to textbox for manual editing  
    see [docs](https://vladmandic.github.io/sdnext-docs/Detailer/) for more information  
  - **Detailer** add option to merge multiple results from each detailer model  
    for example, hands model can result in two hands each being processed separately or both hands can be merged into one composite job  
  - **Control** auto-update width/height on image upload  
  - **Control** auto-determine image save path depending on operations performed  
  - autodetect **V-prediction** models and override default sampler prediction type as needed  
- **SDNQ**  
  - use inference context during quantization  
  - use static compile  
  - rename quantization type for text encoders `default` option to `Same as model`  
- **API**  
  - add `/sdapi/v1/lock-checkpoint` endpoint that can be used to lock/unlock model changes  
    if model is locked, it cannot be changed using normal load or unload methods  
- **Fixes**
  - allow theme type `None` to be set in config  
  - installer dont cache installed state  
  - fix Cosmos-Predict2 retrying TAESD download  
  - better handle startup import errors  
  - fix traceback width preventing copy&paste  
  - fix ansi controle output from scripts/extensions  
  - fix diffusers models non-unique hash  
  - fix loading of manually downloaded diffuser models  
  - fix api `/sdapi/v1/embeddings` endpoint  
  - fix incorrect reporting of deleted and modified files  
  - fix SD3.x loader and TAESD preview  
  - fix xyz with control enabled  
  - fix control order of image save operations  
  - fix control batch-input processing  
  - fix modules merge save model  
  - fix torchvision bicubic upsample with ipex  
  - fix instantir pipeline  
  - fix prompt encoding if prompts within batch have different segment counts  
  - fix detailer min/max size  
  - fix loopback script  
  - fix networks tags display  
  - fix yolo refresh models  
  - cleanup control infotext  
  - allow upscaling with models that have implicit VAE processing  
  - framepack improve offloading  
  - improve prompt parser tokenizer loader  
  - improve scripts error handling  
  - improve infotext param parsing  
  - improve extensions ui search  
  - improve model type autodetection  
  - improve model auth check for hf repos  
  - improve Chroma prompt padding as per recommendations  
  - lock directml torch to `torch-directml==0.2.4.dev240913`  
  - lock directml transformers to `transformers==4.52.4`  
  - improve install of `sentencepiece` tokenizer  
  - add int8 matmul fallback for ipex with onednn qlinear  
- **Refactoring**  
  *note*: none of the removals result in loss-of-functionality since all those features are already re-implemented  
  goal here is to remove legacy code, code duplication and reduce code complexity  
  - obsolete **original backend**  
  - remove majority of legacy **a1111** codebase  
  - remove legacy ldm codebase: `/repositories/ldm`  
  - remove legacy blip codebase: `/repositories/blip`  
  - remove legacy codeformer codebase: `/repositories/codeformer`  
  - remove legacy clip patch model: `/models/karlo`  
  - remove legacy model configs: `/configs/*.yaml`  
  - remove legacy submodule: `/modules/k-diffusion`  
  - remove legacy hypernetworks support: `/modules/hypernetworks`  
  - remove legacy lora support: `/extensions-builtin/Lora`  
  - remove legacy clip/blip interrogate module  
  - remove modern-ui remove `only-original` vs `only-diffusers` code paths  
  - refactor control processing and separate preprocessing and image save ops  
  - refactor modernui layouts to rely on accordions more than individual controls  
  - refactore pipeline apply/unapply optional components & features  
  - split monolithic `shared.py`  
  - cleanup `/modules`: move pipeline loaders to `/pipelines` root  
  - cleanup `/modules`: move code folders used by pipelines to `/pipelines/<pipeline>` folder  
  - cleanup `/modules`: move code folders used by scripts to `/scripts/<script>` folder  
  - cleanup `/modules`: global rename `modules.scripts` to avoid conflict with `/scripts`  
  - override `gradio` installer  
  - major refactoring of requirements and dependencies to unblock `numpy>=2.1.0`  
  - patch `insightface`  
  - patch `facelib`  
  - patch `numpy`  
  - stronger lint rules  
    add separate `npm run lint`, `npm run todo`, `npm run test`, `npm run format` macros
